gpt3_test.go

```diff
@@ -106,6 +106,12 @@ func TestRequestCreationFails(t *testing.T) {
 				return client.SearchWithEngine(ctx, gpt3.AdaEngine, gpt3.SearchRequest{})
 			},
 			"Post \"https://api.openai.com/v1/engines/ada/search\": request error",
+		}, {
+			"Embeddings",
+			func() (interface{}, error) {
+				return client.Embeddings(ctx, gpt3.EmbeddingsRequest{})
+			},
+			"Post \"https://api.openai.com/v1/embeddings\": request error",
```

Comment:
It does not. On line 34 in gpt3_test.go, `TestRequestCreationFails` sets a fake transport on the http client passed to the gpt3 client. The fake transport is a `FakeRoundTripper` (already defined in this code base), which just mocks a transport layer, so no network sockets are opened.

However, these test cases are not testing anything unique. You can see all the same cases are just testing an error response. The error response is exactly the same except for the URL in the error message. This implies the entire set of test cases can be simplified to a single test case that tests the constructed error message meets an expected string.

This is why I claimed in a separate comment that the test cases would benefit from a refactor. In the end, the only thing different for each API call (outside of the streaming completion request) is the struct type passed to functions that are doing the same thing. There's no reason to test the golang JSON unmarshaller or HTTP client over and over. Since all the API methods use the same or copy pasted code (except for the streaming completion endpoint), the code could be refactored, we could test the refactored/reused API call method once and have the same or better confidence that the test cases cover what we need.

The main reason why I did not pick up refactoring the code base and test cases is to keep this PR clean so it's easy to review.